src/ott/neural/methods/flow_matching.py

```diff
@@ -0,0 +1,300 @@
+    loss_fn: Loss function with a signature ``(pred, target) -> loss``.
+    model_callback_fn: Function with a signature ``(model) -> None``, e.g., used
+      to update an :meth:`~ott.neural.networks.velocity_field.EMA` of the model.
+    rngs: Random number generator used for, e.g., dropout, passed to the model.
+
+  Returns:
+    Updates the parameters in-place and returns the loss and the gradient norm.
+  """
+
+  def compute_loss(model: nnx.Module, rngs: nnx.Rngs) -> jax.Array:
+    t, x_t, v_t = batch["t"], batch["x_t"], batch["v_t"]
+    cond = batch.get("cond")
+    v_pred = model(t, x_t, cond, rngs=rngs)
+    return loss_fn(v_pred, v_t).mean()
+
+  loss, grads = nnx.value_and_grad(compute_loss)(model, rngs)
+  optimizer.update(model, grads)
+  grad_norm = optax.global_norm(grads)
+
+  if model_callback_fn is not None:
+    model_callback_fn(model)
+
+  return {"loss": loss, "grad_norm": grad_norm}
+
+
+def evaluate_velocity_field(
+    model: nnx.Module,
+    x: Union[jax.Array, Any],
+    cond: Optional[jax.Array] = None,
+    *,
+    t0: float = 0.0,
+    t1: float = 1.0,
+    reverse: bool = False,
+    num_steps: Optional[int] = None,
+    solver: Optional[diffrax.AbstractSolver] = None,
+    save_trajectory_kwargs: Optional[Dict[str, Any]] = None,
+    save_velocity_kwargs: Optional[Dict[str, Any]] = None,
+    **kwargs: Any,
+) -> diffrax.Solution:
+  """Solve an ODE.
+
+  Args:
+    model: Velocity field with a signature ``(t, x_t, cond) -> v_t``.
+    x: Initial point of shape ``[*dims]``.
+    cond: Condition of shape ``[*cond_dims]``.
+    t0: Start time of the integration.
+    t1: End time of the integration.
+    reverse: Whether to integrate from ``t1`` to ``t0``.
+    num_steps: Number of steps used for solvers with a constant step size.
+    solver: ODE solver. If :obj:`None` and ``step_size = None``,
+      use :class:`~diffrax.Dopri5`. Otherwise use :class:`~diffrax.Euler`.
+    save_velocity_kwargs: Keyword arguments for :class:`~diffrax.SubSaveAt`
+      used to store the velocities along the integration path.
+      The velocity will be saved in :class:`out.ys['v_t'] <diffrax.Solution>`.
+    save_trajectory_kwargs: Keyword arguments for :class:`~diffrax.SubSaveAt`
+      used to store the positions along the integration path.
+      The trajectory will be saved in :class:`out.ys['x_t'] <diffrax.Solution>`.
+    kwargs: Keyword arguments for :func:`~diffrax.diffeqsolve`.
+
+  Returns:
+    The ODE solution.
+  """
+  if isinstance(num_steps, int):
+    step_size = 1.0 / num_steps
+    stepsize_controller = diffrax.ConstantStepSize()
+    solver = diffrax.Euler() if solver is None else solver
+  else:
+    step_size = None
+    stepsize_controller = diffrax.PIDController(rtol=1e-5, atol=1e-5)
+    solver = diffrax.Dopri5() if solver is None else solver
+
+  if reverse:
+    step_size = None if step_size is None else -step_size
+    t0, t1 = t1, t0
+
+  # custom function used when computing the gaussian `
```

Comment:
this comment is not super clear, and it seems the default proposed here is that which is implemented in the only function call using this, maybe this can be simplified?